packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -1412,23 +1408,25 @@ class PubDependencyTree {
     String package, {
     @required Set<String> seen,
     @required Set<String> exclude,
-  }) sync* {
+    List<String> result = const <String>[],
+  }) {
     assert(seen != null);
     assert(exclude != null);
     if (!_dependencyTree.containsKey(package)) {
       // We have no transitive dependencies extracted for flutter_sdk packages
       // because they were omitted from pubspec.yaml used for 'pub upgrade' run.
-      return;
+      return result;
     }
     for (final String dependency in _dependencyTree[package]) {
       if (!seen.contains(dependency)) {
         if (!exclude.contains(dependency)) {
-          yield dependency;
+          result.add(dependency);
```

Comment:
Will it be ok to use `.add` when result is a const empty list (default value).

Reply:
No. I'll change this so the default is null and create one in the method if it's null.